docs/dev/new.rst

```diff
@@ -0,0 +1,113 @@
+
+Building a Conformant Library
+=============================
+
+There are two pieces to building a conformant library:
+
+1. Having a cosmology class that conforms to the Cosmology API
+2. Implementing the required set of modules in the library
+
+Let's start with the 2nd requirement, which is the easier of the two.
+
+.. warning::
+
+    Remember that run-time checks of protocols only validate the existence of
+    attributes and methods, not important details like their input and output
+    types. For more detailed checks it is necessary to inspect the signatures of
+    the methods and attributes.
+
+
+Library Layout
+--------------
+
+At present, the library layout is very simple. All that is required is that the
+library must have a top-level module called "constants".
+
+.. code-block:: text
+
+    library/
+
+        constants.py
+
+
+The :doc:`Cosmology API <../index>` package provides two run-time checkable
+Protocols that can be used for static and run-time checks that the library and the
+constants module conform to the API. See the protocols for details.
+
+- :class:`~cosmology.api.CosmologyNamespace`
+- :class:`~cosmology.api.CosmologyConstantsNamespace`)
```

Comment:
There's an extraneous `)` at the end of this line. Tagging for possible follow-up.

Reply:
Made a followup issues.